File: 01_Data_Parsing/01_Javascript/fileParser.js
const fs = require("fs");
const { parse, stringify } = require("yaml");
const { DOMParser } = require("@xmldom/xmldom"); // install: npm install @xmldom/xmldom

class FileParserCustom {
    parseJson(filePath) {
        const content = fs.readFileSync(filePath, "utf8");
        const data = JSON.parse(content);
        return JSON.stringify(data, null, 4);
    }

    parseXml(filePath) {
        const content = fs.readFileSync(filePath, "utf8");
        const parser = new DOMParser();
        const xmlDoc = parser.parseFromString(content, "text/xml");
        const xmlDocAsString = xmlDoc.toString();
        console.log(xmlDocAsString)

        return xmlDocAsString
    }

    parseYaml(filePath) {
        const content = fs.readFileSync(filePath, "utf8");
        const data = parse(content);
        const dataAsString = stringify(data)
        
        console.log(dataAsString)
        return dataAsString; 
    }

    parseText(filePath) {
        return fs.readFileSync(filePath, "utf8");
    }

    parseCsv(filePath) {
        const content = fs.readFileSync(filePath, "utf8");
        const rows = content.split("\n").map(line => line.split(","));
        return rows.map(row => row.join(", ")).join("\n");
    }
}

module.exports = FileParserCustom;

File: 01_Data_Parsing/01_Python/data_parsing/file_parser.py
import json
import xml.etree.ElementTree as ET
import yaml
import csv

class FileParser:
    def parse_json(self, file_path: str):
        with open(file_path, "r", encoding="utf-8") as f:
            data = json.load(f)
        return data

    def parse_xml(self, file_path: str) -> str:
        tree = ET.parse(file_path)
        root = tree.getroot()

        return ET.tostring(root, encoding="utf-8")

    def parse_yaml(self, file_path: str) -> str:
        with open(file_path, "r", encoding="utf-8") as f:
            data = yaml.safe_load(f)
        return yaml.dump(data, default_flow_style=False)

    def parse_text(self, file_path: str) -> str:
        with open(file_path, "r", encoding="utf-8") as f:
            return f.read()

    def parse_csv(self, file_path: str) -> str:
        with open(file_path, "r", encoding="utf-8") as f:
            reader = csv.reader(f)
            rows = list(reader)

        return "\n".join([", ".join(row) for row in rows])

